src/internal/kopia/wrapper.go

```diff
@@ -389,26 +389,34 @@ func newTreeMap() *treeMap {
 	}
 }
 
-// inflateDirTree returns an fs.Directory tree rooted at the oldest common
-// ancestor of the streams and uses virtualfs.StaticDirectory for internal nodes
-// in the hierarchy. Leaf nodes are virtualfs.StreamingDirectory with the given
-// DataCollections.
+// inflateDirTree returns a set of tags representing all the resource owners and
+// service/categories in the snapshot and a fs.Directory tree rooted at the
+// oldest common ancestor of the streams. All nodes are
+// virtualfs.StreamingDirectory with the given DataCollections if there is one
+// for that node. Tags can be used in future backups to fetch old snapshots for
+// caching reasons.
 func inflateDirTree(
 	ctx context.Context,
 	collections []data.Collection,
 	progress *corsoProgress,
-) (fs.Directory, error) {
+) (fs.Directory, map[string]struct{}, map[string]struct{}, error) {
```

Comment:
This feels a little cluttered.  Thoughts on passing around an internal struct instead?  Ex:
```go
type ownersCats struct {
  resourceOwners map[string]struct{}
  serviceCats map[string]struct{}
}
```
```suggestion
) (fs.Directory, ownersCats, error) {
```